proto/api/registration/registration.proto

```diff
@@ -114,13 +105,15 @@ service Registration {
     rpc ListBySpiffeID(SpiffeID) returns (spire.common.RegistrationEntries);
 
     // Creates an entry in the Federated bundle table to store the mappings of Federated SPIFFE IDs and their associated CA bundle.
-    rpc CreateFederatedBundle(CreateFederatedBundleRequest) returns (spire.common.Empty);
+    rpc CreateFederatedBundle(FederatedBundle) returns (spire.common.Empty);
+    // Retrieves a single federated bundle
+    rpc FetchFederatedBundle(FederatedBundleID) returns (FederatedBundle);
     // Retrieves Federated bundles for all the Federated SPIFFE IDs.
-    rpc ListFederatedBundles(spire.common.Empty) returns (ListFederatedBundlesReply);
+    rpc ListFederatedBundles(spire.common.Empty) returns (FederatedBundles);
     // Updates a particular Federated Bundle. Useful for rotation.
     rpc UpdateFederatedBundle(FederatedBundle) returns (spire.common.Empty);
     // Delete a particular Federated Bundle. Used to destroy inter-domain trust.
-    rpc DeleteFederatedBundle(FederatedSpiffeID) returns (spire.common.Empty);
+    rpc DeleteFederatedBundle(FederatedBundleID) returns (spire.common.Empty);
```

Comment:
Does it make sense to take a FederatedBundle message here with the spiffe id set? There should only be one bundle per trust domain, we can abstract the ID lookup away from the user (if necessary) since it feels kind of cumbersome... same for the FetchFederatedBundle call

Reply:
I was trying to maintain consistency with the other rpc methods that take a SPIFFE ID as input.

Passing in a message with fields that are currently and almost certainly going to be unused feels a bit smelly to me but maybe I am overthinking it.

TBH, I think there is something to be said for uniformity, and I feel like this whole API set could benefit from uniform use of Request/Response messages. One benefit would be a reduction in the number of RPC's. For example, there could be a single List RPC for registration entries with a ListEntriesRequest message containing fields describing filters (possibly via a `oneof` to make the filters mutually exclusive) instead of four different list RPC's.

I'd also be super keen on getting rid of "FederatedBundle" altogether and adding an `id` field to `Bundle`.

I know we need to be careful about backwords compatability so maybe a wholesale revamp is totally out of the question.

Reply:
@azdagron and I discussed this offline. We agreed that consistency is best, and will work to address this and other issues w/ registration api in a v2